src/storage/gcloud-storage/gcp-storage.service.ts

```diff
@@ -119,6 +121,45 @@ export class GCPStorageService {
         return ok(true);
     }
 
+    private async compressFile(file: File): Promise<Buffer> {
+        const chunks: Buffer[] = [];
+
+        for await (const chunk of file.stream()) {
+            chunks.push(chunk as Buffer);
+        }
+        const fileBuffer = Buffer.concat(chunks);
+
+        try {
+            const sharpPipeline = sharp(fileBuffer).resize({
+                width: 2000,
+                height: 2000,
+                fit: 'inside',
+                withoutEnlargement: true,
+            });
+
+            const format = file.type.split('/')[1];
+
+            if (format === 'jpeg' || format === 'jpg') {
+                return sharpPipeline.jpeg({ quality: 75 }).toBuffer();
+            }
+
+            if (format === 'png') {
+                return sharpPipeline.png({ compressionLevel: 9 }).toBuffer();
+            }
+
+            if (format === 'webp') {
+                return sharpPipeline.webp({ quality: 75 }).toBuffer();
+            }
+
+            return fileBuffer;
+        } catch (error) {
+            this.logger.error(
+                'Image compression failed, keeping original file',
+            );
```

Comment:
Should we include the original error / error message somehow to give us an indication of exactly what the error was?

Reply:
Yep good one, fixed in 2d5bfa8df9c69c6c7e780ecf74aa659f422fe639